providers/notificationProvider.tsx

```diff
@@ -0,0 +1,55 @@
+import * as Toast from '@radix-ui/react-toast';
+import type {
+  CSSProperties,
+  Dispatch,
+  FC,
+  PropsWithChildren,
+  SetStateAction,
+} from 'react';
+import { createContext, useEffect, useState } from 'react';
+
+import Notification from '@/components/Common/Notification';
+
+const viewportStyles = {
+  position: 'absolute',
+  bottom: 0,
+  right: 0,
+} satisfies CSSProperties;
+
+type NotificationContextType = {
+  message: string;
```

Comment:
Should this really only allow string? I feel this should allow a React.ReactNode, just because sometimes we might want to render an icon also? or other elements?

Reply:
updated to include elements